### DISCUSSION TITLE
Use Python's multiprocessing.Pool class in many tasks in Airflow

### DISCUSSION THREAD

**Discussion by medina325** [apache/airflow#25606] [Q&A]
Hello Airflow Community!

### Apache Airflow version
2.3.3 (latest released)

### Deployment
docker-compose

### What happened
It hasn't happened yet, I wanna know in advance if I can run into any problems by instantiating processes with the multiprocessing.Pool class in many concurrent tasks in Airflow.
Right now, by testing only one task I can make use of this library to speed up my processing needs, however, I'm curious to know (and kinda scared to try hehe) what will happen if I have, for example, 100 tasks doing the same thing.

Just an observation, following this [question](https://stackoverflow.com/questions/68878031/is-multiprocessing-pool-not-allowed-in-airflow-task-assertionerror-daemonic) in StackOverflow, I set the config ``execute_tasks_new_python_interpreter`` to True in airflow.cfg, otherwise I'd get the ``AssertionError: daemonic processes are not allowed to have children`` error message, which is pretty much the reason why I'm creating this discussion, I wanna avoid unpredictable errors like this to happen (due to my ignorance on some aspects of processes/threads).

**Comment by potiuk** [comment]
This is not airflow problem, this is a problem with Celery way of handling multiprocessing. If you want to use multiprocessing from Celery you need to use billiard instead of multiprocessing https://github.com/celery/billiard (it's developed and maintained by the Celery team). 

It is a drop-in replacement for multiprocessing. You can use it this way:

```
import billiard as multiprocessing
```

**Reply by medina325** [reply]
Nice to know, but I think you misinterpreted what I said. I haven't gotten any errors yet, because I have only used multiprocessing.Pool in one task at a time during their development.
Now I have 100 tasks that are gonna use concurrent processes, and I'm wondering if something can go wrong with my current setup.

**Reply by potiuk** [reply]
Well. You did mention "daemonic process errror" and billiard is response to that.

If you ask if ANYTHING can go wrong yes it can happen with ANY approach. For example if you have 1G memory and run 100 tasks in parallell where each require 100M of memory, then yeah things can go terribly wrong.

But this has nothing to do with airflow, it's just common sense :)

So directly answering your question - yes things can go wrong.

**Reply by medina325** [reply]
Haha yes I agree, that would go terribly wrong. I think my main concern is something like that, extrapolating my resources, by trying to spawn too many processes in my case. Well, guess I'll have to find out soon enough.

Appretiate the patience 👍🏽
